api/grpcserver/node_service.go

```diff
@@ -106,6 +106,30 @@ func (s NodeService) NodeInfo(context.Context, *empty.Empty) (*pb.NodeInfoRespon
 	}, nil
 }
 
+func (s NodeService) PeerInfo(context.Context, *empty.Empty) (*pb.PeerInfoResponse, error) {
+	infos := s.peers.PeerInfo()
+	peers := make([]*pb.PeerInfo, len(infos))
+	// for i, info := range s.peers.PeerInfo() {
+	for i, info := range s.peers.PeerInfo() {
+		connections := make([]*pb.ConnectionInfo, len(info.Connections))
+		for j, c := range info.Connections {
+			connections[j] = &pb.ConnectionInfo{
+				Address:  c.Address.String(),
+				Uptime:   c.Uptime.String(),
+				Outbound: c.Outbound,
+			}
+		}
+		peers[i] = &pb.PeerInfo{
+			Id:          info.ID.String(),
+			Connections: connections,
+			Tags:        info.Tags,
+		}
+	}
+	return &pb.PeerInfoResponse{
```

Comment:
on some nodes we may have up to 10_000 peers. i would like to avoid memory spikes when reading peers from them, if it would be the case then it is better to switch to streaming

Reply:
@dshulyak Switched to streaming now